use seissol v1.3/Rusanov fluxes/ vtkhdf/ extend tpv13

Thomas-Ulrich opened the pull request.

Thomas-Ulrich commented:
I think it is worth updating to the vtkhdf output:
![image](https://github.com/user-attachments/assets/94cbbfe0-b803-4256-a5bc-6d7cb8f6fc93)
![image](https://github.com/user-attachments/assets/6b98f9b0-1a68-4d04-9c90-bff91749fe60)
vs the older output:
![image](https://github.com/user-attachments/assets/3139070a-9623-4950-b945-d6ea10a9dfbd)
![image](https://github.com/user-attachments/assets/5a509c37-7bf9-47a8-a07f-6c0be5d2391b)

AliceGabriel commented:
Thank you Thomas! 
I agree about updating the output.
Question: how different is the output after switching the flux in this coarse mesh?

Thomas-Ulrich commented:
the figure above shows a qualitative answer.

AliceGabriel commented:
Do you mind plotting them in the same figure

AliceGabriel commented:
But overall I agree- we should update this

AliceGabriel commented:
> Do you mind plotting them in the same figure

AliceGabriel commented on tpv13/tpv13_training.geo:
```diff
@@ -32,7 +32,7 @@ Z0 = -42e3;
 domain = newv; Box(domain) = {X0, Y0, Z0, X1-X0, Y1-Y0, -Z0};
 
-// Create the fault as a vertically dipping rectangle, centered in x at the hypocenter
+// Create the fault as a in the x-y plane, centered in x at the hypocenter
```
messed up grammar, missing words

Thomas-Ulrich commented:
Ok, it turns out the 
reader.UpdateTimeStep(5.) that we had first meant read at time simulation time 5+1
(makes not much sense but that what I got when testing).
Anyway, if I put reader.UpdateTimeStep(9.)  and compare with the file 'output/{prefix}-free-surface-4.vtkhdf', I get matching data:

![image](https://github.com/user-attachments/assets/1f2be997-cf3f-4b9a-a101-48c8896efa7b)

Thomas-Ulrich commented:
Latest changes:
- I realized that pv.Plotter() needs to be closed to avoid memory leaks (leading to the RAM getting filled if you play to much with the iterative plots. Now that's fixed.
- Since the last exercise is based on the xdmf output, I reverted to fault xdmf, also because this does not brings much improvement.
- I've added one more step in the rupture speed analysis, in which we decrease initial stress to suppress supershear.

Thomas-Ulrich commented:
Also there is this piece of code that compute sigma_d, sigma_n and R from the stress tensor.
Could be integrated to the notebook for commenting on the initial stress and supershear rupture.
```python
import sympy as sp
import numpy as np

# Define symbolic variables
a, sigma_max, dip, mu_s, mu_d = sp.symbols("a sigma_max dip mu_s mu_d")

# stress tensor
stress = sp.diag((1 + a) * sigma_max / 2, a * sigma_max, sigma_max)
# normal and along dip unit vectors
u_n = sp.Matrix([0, -sp.sin(dip), sp.cos(dip)])
udip = sp.Matrix([0, sp.cos(dip), sp.sin(dip)])

# tractions
traction = stress * u_n

sigma_n = traction.dot(u_n)
print("sigma_n:", sigma_n)
sigma_d = traction.dot(udip)
print("sigma_d:", sigma_d)

R = (sigma_d - mu_d * sigma_n) / ((mu_s - mu_d) * sigma_n)
print("relative prestress ration R:", R)

params = {dip: 60 * np.pi / 180, a: 0.3495, mu_s: 0.54, mu_d: 0.1}
print(
    f"relative prestress ration R evaluated with {params} (nucleation):", R.subs(params)
)

params = {dip: 60 * np.pi / 180, a: 0.3495, mu_s: 0.7, mu_d: 0.1}
print(
    f"relative prestress ration R evaluated with {params} (outside nucleation):",
    R.subs(params),
)

print("sigma_n:", sigma_n.subs(params))
print("sigma_d:", sigma_d.subs(params))
```

davschneller commented on tpv13/parameters.par:
```diff
@@ -34,6 +36,7 @@ printtimeinterval_sec = 1.0                    ! Time interval at which output w
 OutputMask = 1 1 1 1 1 1 1 1 1 1 1             ! turn on and off fault outputs
 refinement_strategy = 2
 refinement = 1
+!vtkorder = 3
```
Commented out?

Thomas-Ulrich replied:
yes, because I decided not to use, because this would require updating the paraview example below.

davschneller reviewed (approved):
LGTM from looking over the source.

(do we even need the `Dockerfile`, i.e. the Frontera container, at the moment? It's not built right now anyways)

Thomas-Ulrich commented:
> (do we even need the `Dockerfile`, i.e. the Frontera container, at the moment? It's not built right now anyways)

you mean, frontera does not need a specific file anymore and we can remove the file?

davschneller commented:
> > (do we even need the `Dockerfile`, i.e. the Frontera container, at the moment? It's not built right now anyways)
> 
> you mean, frontera does not need a specific file anymore and we can remove the file?

It still does need a special one. However, our CI doesn't build it at the moment (that can be fixed quite easily though).
